### PR TITLE
get data from Google API everyday and save a schedule objects

### PR DESCRIPTION
**Pull request by x-Shen**

### CODE REVIEW

File: portalCheckIn/server/route/usersRoute.js [hunk 1]
```diff
@@ -186,73 +23,65 @@ router.post('/checkin', function (req, res) {
         }
         else {
             var newAction = new Action();
-            newAction.type[0] = 'checkin';
+            newAction.type.push = 'checkin';
```
Comment (thireven) [inline portalCheckIn/server/route/usersRoute.js]: This is not how you use push.

File: portalCheckIn/server/route/usersRoute.js [hunk 2]
```diff
@@ -40,42 +23,65 @@ router.post('/checkin', function (req, res) {
...
-                        res.status(201);
-                        res.json({
-                            status: 201,
-                            err : "User is Checked In"
-                        });
-                    }
-                })
+            Action.find({'user._id': newAction.user._id}, {}, {sort: {'createdAt': -1}}, function (actions, err) {
+                
+                if (actions[0].type[0] =='checkout'){
```
Comment (thireven) [inline portalCheckIn/server/route/usersRoute.js]: As mentioned previously, you cannot assume the position of any element in the array

File: portalCheckIn/get_schedule.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+ */
+function storeToken(token) {
+    try {
+        fs.mkdirSync(TOKEN_DIR);
+    } catch (err) {
+        if (err.code != 'EEXIST') {
+            throw err;
+        }
+    }
+    fs.writeFile(TOKEN_PATH, JSON.stringify(token));
```
Comment (thireven) [inline portalCheckIn/get_schedule.js]: This should also be in a try catch block. And should it be a sync?

**Comment by portalconsulting** [comment]
Looking through c3a7784 and then d8d8174, noticed that you changed the crontime twice.
Is there any particular reason?

Apart from that, did you handle the rest of the items that @thireven mentioned?

File: portalCheckIn/server/route/usersRoute.js [hunk 4]
```diff
@@ -40,42 +23,65 @@ router.post('/checkin', function (req, res) {
         }
         else {
             var newAction = new Action();
-            newAction.actionType = 'checkin';
+            newAction.type.push = 'checkin';
```
Comment (thireven) [inline portalCheckIn/server/route/usersRoute.js]: Not how push is used.

Comment (thireven) [inline portalCheckIn/server/route/usersRoute.js]: Has this been tested and confirmed working? I'm seeing a lot of places where things should be failing one way or another. If it hasn't been tested, don't send a pull request.

File: portalCheckIn/get_schedule.js (hunk 3 above)
Comment (x-Shen) [inline portalCheckIn/get_schedule.js]: This is what Google provided in the tutorial. I haven't change anything.

Comment (thireven) [inline portalCheckIn/get_schedule.js]: Do you understand what's going on in that function, then?

Comment (x-Shen) [inline portalCheckIn/get_schedule.js]: I think the function means if there's a token, write it to path, else, catch the error.

**Comment by x-Shen** [comment]
I fixed the errors in push Thi mentioned in the three new commit. And changed find to findOne in get_schedule.

**Comment by thireven** [comment]
I'm going to close this PR as there are too many fundamental issues with the code.

**Pull request closed without merging**